oneTBB cannot be compiled as universal binary on OSX 12.5

Steps to reproduce:
```sh
% cmake -DCMAKE_OSX_ARCHITECTURES="arm64;x86_64" ../
% cmake --build .
```
Compilation errors:
```
[ 41%] Building CXX object src/tbb/CMakeFiles/tbb.dir/itt_notify.cpp.o
[ 43%] Building CXX object src/tbb/CMakeFiles/tbb.dir/main.cpp.o
In file included from /Users/sandye51/Documents/Programming/git_repo/oneTBB/src/tbb/arena.cpp:17:
In file included from /Users/sandye51/Documents/Programming/git_repo/oneTBB/src/tbb/task_dispatcher.h:24:
/Users/sandye51/Documents/Programming/git_repo/oneTBB/src/tbb/scheduler_common.h:230:9: error: always_inline function '_tpause' requires target feature 'waitpkg', but would be inlined into function 'prolonged_pause' that is compiled without support for 'waitpkg'
        _tpause(0, time_stamp + 700);
        ^
[ 46%] Building CXX object src/tbb/CMakeFiles/tbb.dir/market.cpp.o
1 error generated.
make[2]: *** [src/tbb/CMakeFiles/tbb.dir/arena.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 48%] Linking C shared library ../../appleclang_14.0_cxx11_64_relwithdebinfo/libtbbmalloc.dylib
make[1]: *** [src/tbb/CMakeFiles/tbb.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 48%] Built target tbbmalloc
make: *** [all] Error 2

```

--- a/cmake/compilers/AppleClang.cmake
+++ b/cmake/compilers/AppleClang.cmake
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2021 Intel Corporation
+# Copyright (c) 2020-2022 Intel Corporation
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -32,9 +32,15 @@ if (NOT TBB_STRICT AND COMMAND tbb_remove_compile_flag)
 endif()
 
 # Enable Intel(R) Transactional Synchronization Extensions (-mrtm) and WAITPKG instructions support (-mwaitpkg) on relevant processors
-if (NOT "${CMAKE_OSX_ARCHITECTURES}" MATCHES "^arm64$" AND CMAKE_SYSTEM_PROCESSOR MATCHES "(x86_64|amd64|AMD64)") # OSX systems are 64-bit only
+if (CMAKE_OSX_ARCHITECTURES)
+    set(_tbb_target_architectures "${CMAKE_OSX_ARCHITECTURES}")
+else()
+    set(_tbb_target_architectures "${CMAKE_SYSTEM_PROCESSOR}")
+endif()
+if ("${_tbb_target_architectures}" MATCHES "(x86_64|amd64|AMD64)") # OSX systems are 64-bit only
     set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -mrtm $<$<NOT:$<VERSION_LESS:${CMAKE_CXX_COMPILER_VERSION},12.0>>:-mwaitpkg>)
 endif()
+unset(_tbb_target_architectures)
 
 # TBB malloc settings
 set(TBBMALLOC_LIB_COMPILE_FLAGS -fno-rtti -fno-exceptions)

